castle/extractors/headers.py

```diff
@@ -1,27 +1,30 @@
-from castle.headers_formatter import HeadersFormatter
 from castle.configuration import configuration
 
-DEFAULT_BLACKLIST = ['Cookie', 'Authorization']
-DEFAULT_WHITELIST = ['User-Agent']
+ALWAYS_BLACKLISTED = ['Cookie', 'Authorization']
+ALWAYS_WHITELISTED = ['User-Agent']
 
 
 class ExtractorsHeaders(object):
-    def __init__(self, environ):
-        self.environ = environ
-        self.formatter = HeadersFormatter
+    def __init__(self, headers):
+        self.headers = headers
+        self.no_whitelist = len(configuration.whitelisted) == 0
 
     def call(self):
-        headers = dict()
-        has_whitelist = len(configuration.whitelisted) > 0
-
-        for key, value in self.environ.items():
-            name = self.formatter.call(key)
-            if has_whitelist and name not in configuration.whitelisted and name not in DEFAULT_WHITELIST:
-                headers[name] = True
-                continue
-            if name in configuration.blacklisted or name in DEFAULT_BLACKLIST:
-                headers[name] = True
-                continue
-            headers[name] = value
-
-        return headers
+        result = dict()
+
+        for name, value in self.headers.items():
+            result[name] = self._header_value(name, value)
+
+        return result
+
+    def _header_value(self, name, value):
+        if name in ALWAYS_BLACKLISTED:
```

Comment:
this should be `if/elif/else` block

Reply:
actually it was changed to 'if only' on purpose because of return statements and  it makes linter happy